.github/workflows/master.yaml

```diff
@@ -9,46 +9,86 @@ concurrency:
+  # But centos7 jobs have a git version too old and actions/checkout does not 
+  # fully clone the repo, but uses http rest api instead.
+  fetch-version:
+    runs-on: ubuntu-latest
+    # Map the job outputs to step outputs
+    outputs:
+      version: ${{ steps.store_version.outputs.version }}  
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          
+      - name: Install build dependencies
+        run: |
+          sudo apt update 
+          sudo apt install -y cmake build-essential
+      
+      - name: Configure project
+        run: |
+          mkdir build && cd build
+          cmake -DUSE_BUNDLED_DEPS=On ..
+          
+      - name: Load and store Falco version output
+        id: store_version
+        run: |
+          FALCO_VERSION=$(cat build/userspace/falco/config_falco.h | grep 'FALCO_VERSION ' | cut -d' ' -f3 | sed -e 's/^"//' -e 's/"$//')
+          echo "version=${FALCO_VERSION}" >> $GITHUB_OUTPUT     
+
   build-dev-packages:
+    needs: [fetch-version]
     uses: falcosecurity/falco/.github/workflows/reusable_build_packages.yaml@master
     with:
       arch: x86_64
+      version: ${{ needs.fetch-version.outputs.version }}
     secrets: inherit
   
   build-dev-packages-arm64:
+    needs: [fetch-version]
     uses: falcosecurity/falco/.github/workflows/reusable_build_packages.yaml@master
     with:
       arch: aarch64
+      version: ${{ needs.fetch-version.outputs.version }}
     secrets: inherit
     
   publish-dev-packages:
-    needs: [build-dev-packages, build-dev-packages-arm64]
+    needs: [fetch-version, build-dev-packages, build-dev-packages-arm64]
     uses: falcosecurity/falco/.github/workflows/reusable_publish_packages.yaml@master
     with:
       bucket_suffix: '-dev'
-      version: ${{ needs.build-dev-packages.outputs.version }}
+      version: ${{ needs.fetch-version.outputs.version }}
     secrets: inherit
   
   # Both build-dev-docker and its arm64 counterpart require build-dev-packages because they use its output
   build-dev-docker:
-    needs: [build-dev-packages, publish-dev-packages]
+    needs: [fetch-version, build-dev-packages, publish-dev-packages]
     uses: falcosecurity/falco/.github/workflows/reusable_build_docker.yaml@master
     with:
       arch: x86_64
       bucket_suffix: '-dev'
-      version: ${{ needs.build-dev-packages.outputs.version }}
+      version: ${{ needs.fetch-version.outputs.version }}
+      tag: ${{ github.ref_name }}
     secrets: inherit
     
   build-dev-docker-arm64:
-    needs: [build-dev-packages, publish-dev-packages]
+    needs: [fetch-version, build-dev-packages, publish-dev-packages]
     uses: falcosecurity/falco/.github/workflows/reusable_build_docker.yaml@master
     with:
       arch: aarch64
       bucket_suffix: '-dev'
-      version: ${{ needs.build-dev-packages.outputs.version }}
+      version: ${{ needs.fetch-version.outputs.version }}
+      tag: ${{ github.ref_name }}
```

Comment:
I'd use `master` directly here in place of `github.ref_name`; being more explicit is fine and desired given how complex the actions are ;)